src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -141,77 +141,112 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
+                    await consumer.Model.BasicRejectAndRequeueIfOpen(message.DeliveryTag, exclusiveScheduler).ConfigureAwait(false);
 
                     return;
                 }
 
-                using (var tokenSource = new CancellationTokenSource())
+                if (tokenSource.IsCancellationRequested)
                 {
-                    var pushContext = new PushContext(messageId, headers, new MemoryStream(message.Body ?? new byte[0]), new TransportTransaction(), tokenSource, new ContextBag());
-                    await pipe(pushContext).ConfigureAwait(false);
-
-                    if (tokenSource.IsCancellationRequested)
+                    await consumer.Model.BasicRejectAndRequeueIfOpen(message.DeliveryTag, exclusiveScheduler).ConfigureAwait(false);
+                }
+                else
+                {
+                    try
                     {
-                        await Requeue(message.DeliveryTag, messageId).ConfigureAwait(false);
+                        await consumer.Model.BasicAckSingle(message.DeliveryTag, exclusiveScheduler).ConfigureAwait(false);
                     }
-                    else
+                    catch (AlreadyClosedException ex)
                     {
-                        await Acknowledge(message.DeliveryTag, messageId).ConfigureAwait(false);
+                        Logger.Warn($"Failed to acknowledge message '{messageId}' because the channel was closed. The message was returned to the queue.", ex);
                     }
                 }
             }
-            catch (Exception ex)
-            {
-                Logger.Warn($"Error while attempting to process message {messageId}. The message will be requeued.", ex);
-                await Requeue(message.DeliveryTag, messageId).ConfigureAwait(false);
-            }
         }
 
-        async Task Acknowledge(ulong deliveryTag, string messageId)
+        async Task Move(BasicDeliverEventArgs message, string queue, string messageId = null)
         {
             try
             {
-                await consumer.Model.BasicAckSingle(deliveryTag, exclusiveScheduler).ConfigureAwait(false);
+                var channel = channelProvider.GetPublishChannel();
+
+                try
+                {
+                    await channel.RawSendInCaseOfFailure(queue, message.Body, message.BasicProperties).ConfigureAwait(false);
+                }
+                finally
+                {
+                    channelProvider.ReturnPublishChannel(channel);
+                }
             }
-            catch (AlreadyClosedException ex)
+            catch (Exception ex)
             {
-                Logger.Warn($"Attempt to acknowledge message {messageId} failed because the channel was closed. The message will be requeued.", ex);
+                Logger.Error($"Failed to move message{(messageId == null ? "" : " '" + messageId + "'")} to queue '{queue}'. Returning message to original queue...", ex);
```

Comment:
Also, what about using `{(messageId == null ? "" : $" '{messageId}'")}` instead for the message id bit?